core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityTracker.scala

```diff
@@ -169,30 +169,48 @@ class CardinalityTracker(ref: DatasetRef,
   }
 
   /**
-   * Use this method to query the top-k cardinality consumers immediately
-   * under a provided shard key prefix
+   * Use this method to query the top-k cardinality consumers
+   *   under a provided shard key prefix.
    *
-   * @param k
+   * @param depth cardinalities are returned for all prefixes of this size
    * @param shardKeyPrefix zero or more elements that form a valid shard key prefix
    * @return Top-K records, can the less than K if fewer children
    */
-  def topk(k: Int, shardKeyPrefix: Seq[String], addInactive: Boolean): Seq[CardinalityRecord] = {
+  def topk(k: Int, shardKeyPrefix: Seq[String], depth: Int, addInactive: Boolean): Seq[CardinalityRecord] = {
```

Comment:
I wonder if we should get rid of 'k' and use a server configured limit instead. This can remove the priority queue too. The method would be renamed as `scan`. Thoughts ?

Reply:
I like it 100x better than `topk` (unless there's something magic `CardinalityTracker` can do to make `topk` happen super fast, but that doesn't seem to be the case). The caller can find the top k if they want, but that's out-of-scope for the `CardinalityTracker`.

Maybe there shouldn't be a `k` at all? Would this be a good spot for an iterator?

Reply:
I replaced `topk` with `scan(shardKeyPrefix: Seq[String], depth: Int): Seq[CardinalityRecord]`. `NMAX` is configurable within the `CardinalityStore`, and a warning is logged if the result is clipped.